pkg/controller/controller.go

```diff
@@ -431,6 +437,33 @@ func (c *Controller) isScaleOnStarve(
 		len(untaintedNodes) < nodeGroup.Opts.MaxNodes
 }
 
+// scaleOnMaxNodeAge returns true if the node group is at the minimum and there is a node in the untaintedNodes older
+// than the configured node group's maxNodeAge. The idea here is to constantly rotate out the oldest nodes
+// when the node group is at the minimum. This is to ensure the nodes are receiving the most up-to-date configuration
+// from the cloud provider.
+func (c *Controller) scaleOnMaxNodeAge(nodeGroup *NodeGroupState, untaintedNodes []*v1.Node) bool {
+	// Only enable this functionality if the node group has the feature enabled
+	if nodeGroup.Opts.MaxNodeAgeDuration() <= 0 {
+		return false
+	}
+
+	// We don't want to attempt to rotate nodes that have reached the max age if we haven't reached the minimum node
+	// count, as the scaling down of the node group will remove the oldest first anyway.
+	if len(untaintedNodes) != nodeGroup.Opts.MinNodes || len(untaintedNodes) == 0 {
```

Comment:
Should this be <= to catch weird edge cases?

Reply:
No I think we want to keep `len(untaintedNodes) != nodeGroup.Opts.MinNodes`, as we're only wanting this feature to come into play when we're at the minimum node level.

Reply:
What I meant was using `<=` is more defensive in case we (or the cloud provider) have a bug elsewhere in the code that would allow the number of nodes present to be less than the number of MinNodes. In that case, we would actually start checking their age and potentially mess with the scale delta here, which complicates getting back to the MinNodes regardless.

All of this said, it's an unlikely edge case and I'm fine with this.

Reply:
> What I meant was using <= is more defensive in case we (or the cloud provider) have a bug elsewhere in the code that would allow the number of nodes present to be less than the number of MinNodes. 

Exceeding the max and being under the min is already caught by Escalator here - https://github.com/atlassian/escalator/blob/97246b1ff512ecdb000a856f10b673bd1dfcbb8b/pkg/controller/controller.go#L237-L254